libraries/commerce/favorites/actions/toggleFavorites.js

```diff
@@ -5,90 +5,117 @@ import * as pipelines from '../constants/Pipelines';
 import {
   requestAddFavorites,
   requestRemoveFavorites,
-  requestSyncFavorites,
-  receiveSyncFavorites,
-  errorSyncFavorites,
-  idleSyncFavorites,
+  requestAddRemoveFavoritesSync,
+  receiveAddRemoveFavoritesSync,
+  errorAddRemoveFavoritesSync,
   errorFavorites,
+  idleSyncFavorites,
 } from '../action-creators';
 import {
   getProductRelativesOnFavorites,
-  getFavoritesProductsIds,
   getFavoritesCount,
 } from '../selectors/';
 
-const { favorites: { limit = 100 } = {} } = appConfig;
-
-let syncPendingCount = 0;
-let syncInProgress = false;
-
-/**
- * Sends putFavorites requests. Maintains queue and throttling.
- * @returns {Function}
- */
-export const requestSync = () => (dispatch, getState) => {
-  // Syncing in progress. Will request next one on response.
-  if (syncInProgress) {
-    return;
-  }
-  // Nothing to sync. Dispatching idle to trigger stream.
-  if (!syncPendingCount) {
-    dispatch(idleSyncFavorites());
-    return;
-  }
-
-  // Reset pending count since we always send entire list.
-  // Whatever was added before is being synced now.
-  syncPendingCount = 0;
-  syncInProgress = true;
-
-  const state = getState();
-  dispatch(requestSyncFavorites());
+import {
+  REQUEST_ADD_FAVORITES,
+  REQUEST_REMOVE_FAVORITES,
+} from '../constants';
 
-  new PipelineRequest(pipelines.SHOPGATE_USER_PUT_FAVORITES)
-    .setInput({ productIds: getFavoritesProductsIds(state) })
-    .setRetries(0)
-    .dispatch()
-    .then(() => {
-      dispatch(receiveSyncFavorites());
-    })
-    .catch(() => {
-      dispatch(errorSyncFavorites());
-    })
-    .then(() => {
-      syncInProgress = false;
-      dispatch(requestSync());
-    });
-};
+const { favorites: { limit = 100 } = {} } = appConfig;
 
 /**
  * Add favorites action.
  * @param {string} productId Product identifier.
- * @returns {Promise} PipelineRequest dispatch.
+ * @return {Function}
  */
 const addFavorites = productId => (dispatch, getState) => {
   dispatch(requestAddFavorites(productId));
-
   const count = getFavoritesCount(getState());
   if (count > limit) {
     const error = new Error('Limit exceeded');
     error.code = ELIMIT;
     dispatch(errorFavorites(productId, error));
+  }
+};
+/**
+ * Calls Add Favorites pipeline
+ * @param {string} productId Product identifier
+ * @returns {Promise} PipelineRequest dispatch.
+ */
+const syncAddSingleFavorite = productId => (
```

Comment:
I take your point. I don't like addFavorite because it is too similar to another function in the same file called addFavorites which can be confusing in its own right, but also the two do significantly different things which should be reflected in the naming. I chose `dispatchAddFavorite`.

Reply:
Now that you say that, I just realized that you actually created somewhat of a duplicate.
Prefixing this one just introduces inconsistency in naming, adding the prefix does not really help.
What happened here is more complicated than it should be, because figuring out what calls what in which order is almost impossible to tell without digging into it very deeply. Future changes will be pretty complicated and are likely to be underrated. Maybe this already happened because the structure was overcomplicated before already.

Reply:
How do you feel about sendAddFavorite?